tools/autograd/gen_variable_type.py

```diff
@@ -1259,11 +1261,9 @@ def requires_unpack(arg):
             # Okay, we are abusing the definition of 'unpack' here a bit,
             # although it's still getting the non-variable from the variable
             # (in this case via TensorOptions rather than Variable/Tensor).
-            if declaration['use_c10_dispatcher'] == 'full':
-                body.append(UNPACK_OPTIONS.substitute(arg_name=arg['name']))
-            else:
-                assert declaration['use_c10_dispatcher'] == 'with_codegenerated_unboxing_wrapper'
-                body.append(LEGACY_UNPACK_OPTIONS.substitute(arg_name=arg['name']))
+            assert declaration['use_c10_dispatcher'] == 'with_codegenerated_unboxing_wrapper', \
+                "VariableKernel shouldn't take TensorOptions if the op is c10-full"
+            body.append(LEGACY_UNPACK_OPTIONS.substitute(arg_name=arg['name']))
```

Comment:
Probably time to rename `LEGACY_UNPACK_OPTIONS`?

Reply:
renamed it to `LEGACY_WRAP_OPTIONS`. It's still legacy since it's only for non-c10-full ops. Do you have a better idea?